robowflex_library/src/robot.cpp

```diff
@@ -369,28 +369,28 @@ bool Robot::loadKinematics(const std::string &group_name)
     const auto &groups = kinematics_->getKnownGroups();
     if (groups.empty())
     {
-        RBX_WARN("No kinematics plugins defined. Fill and load kinematics.yaml!");
+        RBX_ERROR("No kinematics plugins defined. Fill and load kinematics.yaml!");
         return false;
     }
 
     if (!model_->hasJointModelGroup(group_name))
     {
-        RBX_WARN("No JMG defined for `%s`!", group_name);
+        RBX_ERROR("No JMG defined for `%s`!", group_name);
         return false;
     }
 
     const auto &subgroups = model_->getJointModelGroup(group_name)->getSubgroupNames();
-    // if no subgroups exist use the given group name.
-    auto group_names = subgroups.empty() ? std::vector<std::string>{group_name} : subgroups;
+    std::vector<std::string> load_names(subgroups);
+    load_names.push_back(group_name);
```

Comment:
You should check if the top level group has a kinematics solver. See the check inside the loop where it verifies if it is in the known groups.

Reply:
Lines 389-390 in this pull